core/include/debug.h

```diff
@@ -92,10 +94,7 @@ extern "C" {
  *
  * @note Another name for ::DEBUG_PRINT
  */
-#define DEBUG(...) DEBUG_PRINT(__VA_ARGS__)
-#else
-#define DEBUG(...)
-#endif
+#define DEBUG(...) if (ENABLE_DEBUG) DEBUG_PRINT(__VA_ARGS__)
```

Comment:
Confirmed that Clang (Apple LLVM version 7.3.0 (clang-703.0.29)) and arm-none-eabi-gcc (GCC) 5.1.0 eliminates `if (0) { ... }` block with `-O0` optimization level.